meta-mender-core/classes/mender-bootstrap-artifact.bbclass

```diff
@@ -30,33 +45,33 @@ IMAGE_CMD:bootstrap-artifact() {
             extra_args="$extra_args -k ${MENDER_ARTIFACT_SIGNING_KEY}"
         fi
 
-        if [ -n "${MENDER_ARTIFACT_NAME_DEPENDS}" ]; then
+        if [ -n "${MENDER_ARTIFACT_NAME_DEPENDS_BOOTSTRAP}" ]; then
             cmd=""
-            apply_arguments "--artifact-name-depends" "${MENDER_ARTIFACT_NAME_DEPENDS}"
+            apply_arguments "--artifact-name-depends" "${MENDER_ARTIFACT_NAME_DEPENDS_BOOTSTRAP}"
             extra_args="$extra_args $cmd"
         fi
 
-        if [ -n "${MENDER_ARTIFACT_PROVIDES}" ]; then
+        if [ -n "${MENDER_ARTIFACT_PROVIDES}" -o -n "${MENDER_ARTIFACT_PROVIDES_BOOTSTRAP}" ]; then
```

Comment:
Here, and everywhere below, I don't understand why you decided to not use the `_FINAL` variables directly, like:
```suggestion
        if [ -n "${MENDER_ARTIFACT_PROVIDES_FINAL}" ]; then
```
Both would work, but I think using FINAL would be more robust against future changes in the definitions above or other variable expansion bugs.

Reply:
"-n" checks for the length of the argument, due to how final is defined I think "-n" would always return true since the space between PROVIDES and PROVIDES_BOOTSTRAP would always be present even if both variables are undefined

![image](https://github.com/mendersoftware/meta-mender/assets/111505340/696b6e58-8c5e-46f1-aab1-4a9007975d36)

There might be a way to do this otherwise in yocto that I'm not aware of

Reply:
Good point!

I was trying to prevent a hacky user defining `MENDER_ARTIFACT_PROVIDES_FINAL=""` in his/her local.conf and that would defeat the check. But these being internal variables for the class we don't really need to be that paranoid. Thanks for proving me wrong :+1: